internal/profiles/service.go

```diff
@@ -276,6 +291,63 @@ func (p *profileService) UpdateProfile(
 	return profile, nil
 }
 
+func (p *profileService) PatchProfile(
+	ctx context.Context,
+	projectID uuid.UUID,
+	profileID uuid.UUID,
+	subscriptionID uuid.UUID,
+	patch *minderv1.Profile,
+	updateMask *fieldmaskpb.FieldMask,
+	qtx db.Querier,
+) (*minderv1.Profile, error) {
+	oldProfilePb, err := getProfilePBFromDB(ctx, profileID, projectID, qtx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get profile: %w", err)
+	}
+
+	patchProfilePb(oldProfilePb, patch, updateMask)
+
+	return p.UpdateProfile(ctx, projectID, subscriptionID, oldProfilePb, qtx)
+}
+
+func patchProfilePb(oldProfilePb, patchPb *minderv1.Profile, updateMask *fieldmaskpb.FieldMask) {
+	// if there is no update mask, return the old profile. The grpc-rest gateway always sets the update mask
```

Comment:
This is confusing, as it says "return", but there's no return value.  Maybe you mean "an empty field mask means apply no changes"?